src/Native/Unix/configure.cmake

```diff
@@ -321,6 +321,85 @@ check_function_exists(
     mach_timebase_info
     HAVE_MACH_TIMEBASE_INFO)
 
+set (PREVIOUS_CMAKE_REQUIRED_FLAGS ${CMAKE_REQUIRED_FLAGS})
+set (CMAKE_REQUIRED_FLAGS "-Werror -Wsign-conversion")
+
+check_cxx_source_compiles(
+    "
+    #include <sys/socket.h>
+
+    int main()
+    {
+        int fd;
+        sockaddr* addr;
+        socklen_t addrLen;
+
+        int err = bind(fd, addr, addrLen);
+        return 0;
+    }
+    "
+    BIND_ADDRLEN_UNSIGNED
+)
+
+check_cxx_source_compiles(
+    "
+    #include <sys/socket.h>
+
+    int main()
+    {
+        int fd;
+        sockaddr* addr;
+        int32_t addrLen;
+
+        int err = bind(fd, addr, addrLen);
+        return 0;
+    }
+    "
+    BIND_ADDRLEN_SIGNED
+)
+
+if(NOT BIND_ADDRLEN_UNSIGNED AND NOT BIND_ADDRLEN_SIGNED)
+    message(FATAL_ERROR "The addrlen parameter in bind() must be a signed int or an unsigned int.")
+endif()
```

Comment:
I don't think we need to have this error or the BIND_ADDRLEN_SIGNED test.  If the unsigned test fails, then we'll try to compile the src with the signed one, and it'll fail, at which point we can address it based on what alternative is needed.  Same for the case below.